src/Persistence/Reflection/EnumReflectionProperty.php

```diff
@@ -30,6 +32,30 @@ public function __construct(ReflectionProperty $originalReflectionProperty, stri
         $this->enumType                   = $enumType;
     }
 
+    /** {@inheritDoc} */
+    public function getDeclaringClass(): ReflectionClass
+    {
+        return $this->originalReflectionProperty->getDeclaringClass();
+    }
+
+    /** {@inheritDoc} */
+    public function getName(): string
```

Comment:
you should be adding the `@psalm-immutable` annotation (which is not the same than `@psalm-pure` that you tried)

Reply:
I tried it locally, to no avail, then noticed I still had Psalm 4, apparently because Psalm 5 is not compatible with php 8.3 yet, so I just tried pushing, and it doesn't work with Psalm 5 either.

Reply:
https://psalm.dev/r/f5566261cf :thinking:

Reply:
Just `@psalm-external-mutation-free` seems sufficient.